app/models/envelope.rb

```diff
@@ -60,25 +61,30 @@ class Envelope < ActiveRecord::Base
   NOT_FOUND = 'Envelope not found'.freeze
   DELETED = 'Envelope deleted'.freeze
 
-  def self.by_resource_custom_id(field, id)
-    find_by('processed_resource @> ?', { field => id }.to_json)
-  end
-
   def self.by_resource_id(id)
-    find_by('processed_resource @> ?', { '@id' => id }.to_json)
+    return unless id
+
+    where(
+      'processed_resource @> ?',
```

Comment:
maybe this query isn't using the lower index (we can check with `explain`). The `@>` is a "contains" matcher, it will check if a json chunk is contained on another, usually we build a `gin` index for handling data on it . 
But the `processed_resource->>'@id'` is direct accessor to the value (parsed to '::text'), the `lower` index was made on this, so probably we should use it .
I'm not sure if postgresql is that magical to use the `->>` index on the `@>` matcher , would be crazy if it does.

Can you run explain on both, for us to check?
```
find_by('processed_resource @> ?', {'@id': id.downcase}.to_json).explain

find_by("processed_resource->>'@id' = ?", id.downcase).explain
```

For example, on unbounded we don't have an index for direct access, but we do have a GIN on the jsonb:
```
[2] pry(main)> Resource.where("metadata->>'grade' = ?", 'grade 6').explain
  Resource Load (4.5ms)  SELECT "resources".* FROM "resources" WHERE (metadata->>'grade' = 'grade 6')
=> EXPLAIN for: SELECT "resources".* FROM "resources" WHERE (metadata->>'grade' = 'grade 6')
                          QUERY PLAN
--------------------------------------------------------------
 Seq Scan on resources  (cost=0.00..413.87 rows=32 width=594)
   Filter: ((metadata ->> 'grade'::text) = 'grade 6'::text)
(2 rows)




[3] pry(main)> Resource.where("metadata @> ?", {grade: 'grade 6'}.to_json).explain
  Resource Load (2.6ms)  SELECT "resources".* FROM "resources" WHERE (metadata @> '{"grade":"grade 6"}')
=> EXPLAIN for: SELECT "resources".* FROM "resources" WHERE (metadata @> '{"grade":"grade 6"}')
                                        QUERY PLAN
-------------------------------------------------------------------------------------------
 Bitmap Heap Scan on resources  (cost=12.05..33.65 rows=6 width=594)
   Recheck Cond: (metadata @> '{"grade": "grade 6"}'::jsonb)
   ->  Bitmap Index Scan on index_resources_on_metadata  (cost=0.00..12.05 rows=6 width=0)
         Index Cond: (metadata @> '{"grade": "grade 6"}'::jsonb)
(4 rows)

```